src/Schema.php

```diff
@@ -121,6 +121,15 @@ private static function createOrPrependFile($pathname, $template, $separator = '
     else {
       $content = file_get_contents($pathname);
       if (FALSE === strpos($content, $template)) {
+        // Remove old content if present, looking for the template block's
+        // BEGIN and END lines.
+        $templateLines = explode("\n", rtrim($template, "\n"));
+        $start = strpos($content, $templateLines[0]);
+        $end = strpos($content, end($templateLines));
+        if ($start !== FALSE && $end !== FALSE) {
+          $content = substr_replace($content, '', $start, $end - $start + 1);
+        }
+
         $content = $template . $separator . $content;
         $write_content = TRUE;
```

Comment:
1. Let's update the existing snippet instead of removing it and inserting the new one at a different position.
1. Let's be consistent with begin/end (and don't mix start into it).
1. Local variable names should be in snake_case.
```suggestion
        // Remove old content if present, looking for the template block's
        // BEGIN and END lines.
        $template_lines = explode("\n", rtrim($template, "\n"));
        $begin = strpos($content, $template_lines[0]);
        $end = strpos($content, end($template_lines));
        if ($begin !== FALSE && $end !== FALSE) {
          $content = substr_replace($content, $template, $begin, $end - $begin + 1);
        }
        else {
          $content = $template . $separator . $content;
        }
        $write_content = TRUE;
```